.github/workflows/on.pull-request.main.yaml

```diff
@@ -45,3 +45,38 @@ jobs:
           make build-controller SERVICE=$SERVICE
         env:
           SERVICE: ${{ matrix.service }}
+
+  test-controllers:
+    name: test controllers
+    strategy:
+      fail-fast: false
+      matrix:
+        service:
+          - s3
+          - ecr
+          - sns
+          - sqs
+          - elasticache
+          - dynamodb
+          - apigatewayv2
+    runs-on: self-hosted
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
+        id: go
+      - name: checkout code
+        uses: actions/checkout@v2
+      - name: Clean up stale docker images
+        run: docker image prune -f
```

Comment:
Is the above necessary? All images created via the `scripts/build-controller-image.sh` call in the `scripts/kind-build-test.sh` script should produce unique images that are properly layered, so unless I am mistaken, the above isn't necessary.

Unless, of course, you're doing this in order to reduce the memory footprint of the self-hosted runners?

Reply:
Yes so are right. Not needed. I was trying to cleanup but realised later each image is unique.